services/core/shared/core/compat/sdk_v5/blockchainIndex.js

```diff
@@ -130,124 +140,153 @@ const getGeneratorPkInfoArray = async (blocks) => {
+const voteAggregatesQueue = Queue('votingQueue', updateVoteAggregates, 1);
+const blockRewardsQueue = Queue('blockRewardsQueue', updateBlockRewards, 1);
+
+const ensureArray = (e) => Array.isArray(e) ? e : [e];
+
+const indexBlock = async job => {
+	const { height } = job.data;
 
 	const blocksDB = await getBlocksIndex();
-	const blocks = await getBlocksByHeightBetween(fromHeight, toHeight);
+	const blocks = await getBlockByHeight(height);
 	const connection = await getDbConnection();
 	const trx = await startDbTransaction(connection);
 
-	const transactionStartLogMessage = fromHeight === toHeight
-		? `Created new MySQL transaction to index block at height ${fromHeight}`
-		: `Created new MySQL transaction to index blocks in height range ${fromHeight}-${fromHeight}`;
-	logger.debug(transactionStartLogMessage);
+	logger.debug(`Created new MySQL transaction to index block at height ${height}`);
 
-	if (!validateBlocks(blocks)) throw new Error(`Error: Invalid blocks ${fromHeight}-${toHeight} }`);
+	if (!validateBlocks(blocks)) throw new Error(`Error: Invalid block ${height} }`);
 	try {
-		const accountsDB = await getAccountsIndex();
 		const transactionsDB = await getTransactionsIndex();
 		const votesDB = await getVotesIndex();
 		const multisignatureDB = await getMultisignatureIndex();
-		const generatorPkInfoArray = await getGeneratorPkInfoArray(blocks);
 
-		const accountsByPublicKey = await getAccountsByPublicKey(generatorPkInfoArray);
-		const { votes, votesToAggregateArray } = await getVoteIndexingInfo(blocks);
 		const {
 			accounts: accountsFromTransactions,
 			transactions,
 			multisignatureInfoToIndex,
 		} = await getTransactionIndexingInfo(blocks);
 
-		const allAccounts = accountsByPublicKey.concat(accountsFromTransactions);
-		if (allAccounts.length) await accountsDB.upsert(allAccounts, trx);
+		const generatorPkInfoArray = await getGeneratorPkInfoArray(blocks);
+		const accountsByPublicKey = await getAccountsByPublicKey(generatorPkInfoArray);
+		ensureArray(accountsByPublicKey).forEach(a => indexAccountWithData(a));
+		ensureArray(accountsFromTransactions).forEach(a => indexAccountWithData(a));
+
 		if (transactions.length) await transactionsDB.upsert(transactions, trx);
 		if (multisignatureInfoToIndex.length) await multisignatureDB
 			.upsert(multisignatureInfoToIndex, trx);
-		if (votes.length) await votesDB.upsert(votes, trx);
 
-		await updateProducedBlocksAndRewards(generatorPkInfoArray, trx);
-		await updateVoteAggregates(votesToAggregateArray, trx);
+		if (KEY_BASED_ACCOUNT_UPDATE === true) {
+			const addresses = ensureArray(accountsFromTransactions)
+				.filter(a => a.publicKey).map(a => a.publicKey);
+			const publicKeys = ensureArray(accountsFromTransactions)
+				.filter(a => !a.publicKey).map(a => a.address);
+
+			blocks.forEach(block => indexAccountByPublicKey(block.generatorPublicKey));
+			publicKeys.forEach(pk => indexAccountByPublicKey(pk));
+			addresses.forEach(a => indexAccountByAddress(a));
+		}
```

Comment:
This code-block is never executed since `KEY_BASED_ACCOUNT_UPDATE` is defined as a constant and defaults to `false`.

Reply:
That is correct, as it does the same as the logic above, but it does not entirely work. I would like to fix that code in the `0.7.x` branch.